pkg/console/hmy.go

```diff
@@ -0,0 +1,296 @@
+	if err != nil {
+		return nil, err
+	}
+	if tempLimit < 0 {
+		return nil, errors.New(fmt.Sprintf("gas-limit can not be negative: %s", gasLimit))
+	}
+	gLimit := uint64(tempLimit)
+
+	amt, err := common.NewDecFromString(amount)
+	if err != nil {
+		return nil, fmt.Errorf("amount %w", err)
+	}
+
+	gPrice, err := common.NewDecFromString(gasPrice)
+	if err != nil {
+		return nil, fmt.Errorf("gas-price %w", err)
+	}
+
+	toP := &to
+	if to == "" {
+		toP = nil
+	}
+
+	nonce := transaction.GetNextPendingNonce(from, networkHandler)
+	err = ctrlr.ExecuteTransaction(
+		nonce, gLimit,
+		toP,
+		uint32(b.console.shardId), uint32(b.console.shardId),
+		amt, gPrice,
+		[]byte{},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return call.VM.ToValue(*ctrlr.TransactionHash()), nil
+}
+
+func (b *bridge) HmyLockAccount(call jsre.Call) (goja.Value, error) {
+	address := call.Arguments[0].String()
+
+	_, _, err := store.LockKeystore(address)
+	if err != nil {
+		return nil, err
+	}
+
+	return goja.Null(), nil
+}
+
+func (b *bridge) HmyImportRawKey(call jsre.Call) (goja.Value, error) {
+	privateKey := call.Arguments[0].String()
+	password := call.Arguments[1].String()
+
+	name, err := account.ImportFromPrivateKey(privateKey, "", password)
+	if err != nil {
+		return nil, err
+	}
+
+	return call.VM.ToValue(name), nil
+}
+
+func (b *bridge) HmyUnlockAccount(call jsre.Call) (goja.Value, error) {
+	if len(call.Arguments) < 3 {
+		return nil, errors.New("arguments < 3")
+	}
+	address := call.Arguments[0].String()
+	password := call.Arguments[1].String()
+	unlockDuration := call.Arguments[2].ToInteger()
+
+	_, _, err := store.UnlockedKeystoreTimeLimit(address, password, time.Duration(unlockDuration)*time.Second)
+	if err != nil {
+		return nil, err
+	}
+
+	return goja.Null(), nil
+}
+
+func (b *bridge) HmyNewAccount(call jsre.Call) (goja.Value, error) {
+	return goja.Null(), nil
+}
+
+func (b *bridge) HmySign(call jsre.Call) (goja.Value, error) {
+	dataToSign := call.Arguments[0].String()
+	addressStr := call.Arguments[1].String()
+	password := call.Arguments[2].String()
+
+	ks := store.FromAddress(addressStr)
+	if ks == nil {
+		return nil, fmt.Errorf("could not open local keystore for %s", addressStr)
+	}
+
+	acc, err := ks.Find(accounts.Account{Address: address.Parse(addressStr)})
+	if err != nil {
+		return nil, err
+	}
+
+	message, err := signMessageWithPassword(ks, acc, password, []byte(dataToSign))
+	if err != nil {
+		return nil, err
+	}
+
+	return call.VM.ToValue(hex.EncodeToString(message)), nil
+}
+
+func signMessageWithPassword(keyStore *keystore.KeyStore, account accounts.Account, password string, data []byte) (sign []byte, err error) {
+	signData := append([]byte("\x19Ethereum Signed Message:\n" + strconv.Itoa(len(data))))
+	msgHash := hash.Keccak256(append(signData, data...))
+
+	sign, err = keyStore.SignHashWithPassphrase(account, password, msgHash)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(sign) != 65 {
```

Comment:
![image](https://user-images.githubusercontent.com/14296262/121914010-78ad9e80-cd64-11eb-95c0-d94f4fcaaf2f.png)

The returned by the secp256k1 algorithm is 65 bytes, Here is just a check to prevent panic